test/test_helper.rb

```diff
@@ -64,6 +64,20 @@ def teardown
       @logger_stream.close
     end
 
+    def assert_deploy_status(desired_status, result)
```

Comment:
Since the result of deploy is boolean, there's not much value in introducing statuses like `failure` and `success`, especially if there are only two of them. WDYT about helpers like `assert_deploy_succeeds` and `assert_deploy_fails`?

Reply:
I like that suggestion. What I was trying to avoid was `assert_deploy_status(false, result)`, which I felt read poorly.